src/lib.rs

```diff
@@ -69,93 +145,69 @@ pub struct ResolutionResponse {
 )]
 async fn resolve_did(
     Path(arg): Path<String>,
-    State(resolver): State<SharedResolver>,
+    State(clients): State<NetworkClients>,
 ) -> Result<Json<ResolutionResponse>, (StatusCode, String)> {
     let did = IotaDID::parse(&arg).map_err(|e| (StatusCode::BAD_REQUEST, e.to_string()))?;
-    let resolved = resolver
-        .resolve(&did)
+    let network = did.network_str().to_string();
+
+    let object_id = ObjectID::from_str(did.tag_str()).map_err(|e| (StatusCode::BAD_REQUEST, e.to_string()))?;
+
+    let client = clients
+        .get(&network)
+        .ok_or_else(|| (StatusCode::BAD_REQUEST, format!("Unsupported network: {}", network)))?;
+
+    let identity = get_identity(client, object_id)
         .await
-        .map_err(|e| match e.error_cause() {
-            ErrorCause::HandlerError { source, .. }
-                if source
-                    .source()
-                    .is_some_and(|e| e.to_string().contains("not found")) =>
-            {
-                (
-                    StatusCode::NOT_FOUND,
-                    "The requested DID document was not found".to_owned(),
-                )
-            }
-            _ => (StatusCode::INTERNAL_SERVER_ERROR, e.to_string()),
+        .map_err(|e| (StatusCode::INTERNAL_SERVER_ERROR, e.to_string()))?
+        .ok_or_else(|| {
+            (
+                StatusCode::NOT_FOUND,
+                "The requested DID document was not found".to_owned(),
+            )
         })?;
```

Comment:
Taking back what I said in a previous review comment. Unfortunately we can't just use `get_identity` as that only queries `OnChainIdentity`s and doesn't for instance resolve unmigrated aliases or migrated identities. We gotta go back to the previous implementation using `client.resolve_did` but making sure we return a 404 when the document is not found